src/Sentry/Internal/SynchronizedList.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Sentry.Internal
+{
+    // Inefficient, but okay for our use case
+    internal class SynchronizedList<T> : IList<T>
+    {
+        private readonly object _lock = new object();
+        private readonly List<T> _innerList;
+
+        public int Count
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _innerList.Count;
+                }
+            }
+        }
+
+        public bool IsReadOnly => false;
+
+        public SynchronizedList(List<T> innerList) =>
+            _innerList = innerList;
+
+        public SynchronizedList() : this(new List<T>()) {}
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            lock (_lock)
+            {
+                return _innerList.GetEnumerator();
```

Comment:
This enumerator now potentially going to break when the underlying collection mutates.
The safe approach here would be to do return `new List(_innerList)`.